File: python/gsc/conf/inbound/mixed.py
from typing import Literal, Optional

import pydantic


class Mixed(pydantic.BaseModel):
    type: Literal["mixed"] = "mixed"
    tag: str = "in-mixed"

    # Listen Fields
    listen: str = "127.0.0.1"
    listen_port: Optional[int] = 2080

    class User(pydantic.BaseModel):
        username: str
        password: str

    users: Optional[list[User]] = None

File: src/internal/singbox/conf/inbound/mixed.rs
use serde::{Deserialize, Serialize};

use crate::internal::singbox::conf::shared::listen::Listen;

#[serde_with::skip_serializing_none]
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
pub struct Mixed {
    pub tag: String,
    #[serde(flatten)]
    pub listen: Listen,
    pub users: Option<Vec<User>>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct User {
    pub username: String,
    pub password: String,
}
